pytradfri/device/light.py

```diff
@@ -27,69 +48,70 @@ class Light:
     pdf
     """
 
-    def __init__(self, device, index):
+    def __init__(self, device: Device, index: int):
         """Create object of class."""
         self.device = device
         self.index = index
 
     @property
-    def supported_features(self):
+    def supported_features(self) -> int:
         """Return supported features."""
-        return supported_features(self.raw)
+        return supported_features(self.raw.dict())
 
     @property
-    def state(self):
+    def state(self) -> bool:
         """Return device state."""
-        return self.raw.get(ATTR_DEVICE_STATE) == 1
+        return self.raw.state == 1
 
     @property
-    def dimmer(self):
+    def dimmer(self) -> int | None:
         """Return dimmer if present."""
         if self.supported_features & SUPPORT_BRIGHTNESS:
```

Comment:
dimmer is not optional in the response model. Can it still be not supported?

Reply:
I can't think of any light device that is not dimmable. Perhaps we should drop this supported_features-check and just return the int?

Reply:
Yeah, maybe. Ok to leave for the future though.